### PR TITLE
feat(eslint-plugin-jsx-a11y): Upgrade eslint and eslint-plugin-jsx-a11y

### PR DESCRIPTION
**Pull request by jamsea**
Upgrade eslint and related plugins. Migrate changes from NFL's settings

New rules were enabled so lint failures are likely

### CODE REVIEW

File: legacy.js [hunk 1]
```diff
@@ -14,7 +14,6 @@ module.exports = {
         mocha: false,
         jasmine: false
     },
-    ecmaFeatures: {},
```
Comment (potench) [inline legacy.js]: Was this object deprecated or are you deleting because it's empty?

File: package.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "eslint-config-nfl",
-  "version": "11.1.0",
+  "version": "12.0.0",
   "description": "NFL's ESLint config",
   "main": "index.js",
   "scripts": {
+    "changelog": "conventional-changelog -p angular -i CHANGELOG.md -s",
```
Comment (potench) [inline package.json]: Noticed you pretty-fied the JSON here. Should you provide a changelog for this project as well?

File: legacy.js (hunk 1 above)
Comment (jamsea) [inline legacy.js]: It was moved to `parserOptions` http://eslint.org/docs/user-guide/configuring

File: package.json (hunk 2 above)
Comment (jamsea) [inline package.json]: I can do that 👍

**Pull request merged**